experiments/snippets/smartAdopt/README.md

```diff
@@ -0,0 +1,19 @@
+# Smart Adopt
+Smart adopt is a library capable of intelligently adopting records and fields between different models. There are many different use cases for implementing this library, from mass-creating junction records to simplifying a 'multi-select' experience. It's capable of a number of things including updating existing records which match between 2 models, create new records in the destination model where there are unmatched records, or mark records for deletion in the destination model where none exists in the source. It can also specify key fields arbitrarily between the 2 models and selectively choose which data fields are or are not included in a given upsert operation.
```

Comment:
So my biggest comment that this seems like one of those things that, once you hit the use case, this is snippet 10/10 amazing. But I'll be honest, I'm having a little trouble parsing use cases here because I'm not as familiar working with junction objects and the like. 

I think some of this comes down to the structure of the intro. In one paragraph we try to convey the intent, the features, the use cases, and it all kind of meshes together for unfamiliar. I say let's split this up.

```
# Smart Adopt

This library intelligently adopts records, or fields within records, across models.

## Use Cases

Some of the primary use cases involve mass-creating junction objects and simplifying multi-select experiences. For example ... (Having more info here would be awesome. Maybe some sort of concrete use case?)

## Features

(let's go with a bullet list here to make this easier to glance through)

- Updating existing records which match between 2 models
- Create new records in the destination model where there are unmatched records
- Mark records for deletion in the destination model where none exists in the source. 
- Arbitrarily specify key fields between 2 models 
- Selectively choose which data fields are included in a given operation

## Installation
...
```

(As a side note, you say `upsert` in the original text. I'd either avoid using that or update the parameters to utilize that language.)